libevmasm/Assembly.cpp

```diff
@@ -298,6 +416,43 @@ Json::Value Assembly::assemblyJSON(map<string, unsigned> const& _sourceIndices,
 	return root;
 }
 
+shared_ptr<Assembly> Assembly::loadFromAssemblyJSON(Json::Value const& _json, vector<string> const& _sourceList /* = {} */, bool _isCreation /* = true */)
+{
+	if (!_json[".code"].isArray())
+		return {};
+
+	shared_ptr<Assembly> result = make_shared<Assembly>(_isCreation, "");
+	vector<string> sourceList;
+	if (_sourceList.empty())
+	{
+		if (_json.isMember("sourceList"))
+			for (auto const& it: _json["sourceList"])
+				sourceList.emplace_back(it.asString());
+	}
+	else
+		sourceList = _sourceList;
+	result->setSources(sourceList);
+	result->addAssemblyItemsFromJSON(_json[".code"]);
+	if (_json[".auxdata"].isString())
+		result->m_auxiliaryData = fromHex(_json[".auxdata"].asString());
+	Json::Value const& data = _json[".data"];
+	for (Json::ValueConstIterator itr = data.begin(); itr != data.end(); itr++)
+	{
+		solAssert(itr.key().isString(), "");
+		string key = itr.key().asString();
+		Json::Value const& code = data[key];
+		if (code.isString())
+			result->m_data[h256(fromHex(key))] = fromHex(code.asString());
+		else
+		{
+			shared_ptr<Assembly> subassembly(Assembly::loadFromAssemblyJSON(code, sourceList, /* isCreation = */ false));
+			assertThrow(subassembly, AssemblyException, "");
+			result->m_subs.emplace_back(make_shared<Assembly>(*subassembly));
```

Comment:
Could `m_subs` be a vector of unique_ptrs? If it can, it should, since from what I can tell, we're only iterating over its contents later on, so no need for this to be a sharted_ptr.

Reply:
Maybe that's possible to change that, but it would be an unrelated change. But if it makes sense, we could create another PR changing this.